docs/docs/glossary.md

```diff
@@ -2,7 +2,272 @@
+
+### Components
+
+Components are independent and re-usable chunks of code powered by [React](#react) that, when combined, make up your website or app.
+
+A component can include components within it. In fact, [pages](#page) and [templates](#templates) are examples of components.
+
+### Config
+
+The configuration file, `gatsby-config.js` tells Gatsby information about your website. A common option set in config is your sites meta data that can power your SEO meta tags.
+
+## D
+
+### Data Source
+
+A source of data that usually gets fed into Gatsby using [source plugins](#source-plugin). A data source is often a [Headless CMS](#headless-cms), but it could also include Markdown, JSON, or YAML files.
+
+### Database
+
+A database is a structured collection of data or content. Often a [CMS](#cms) will save to a database using [back-end technologies](#back-end). They're often accessed in Gatsby via a [source plugin](#source-plugin)
+
+### Decoupled
+
+Decoupling describes the separation of different concerns. With [Gatsby](#gatsby) this most commonly means decoupling the [Front-End](#front-end) from the [Back-End](#back-end)
+
+### Deploy
+
+The process of [building](#build) your website or app and uploading onto a [hosting provider](#hosting).
+
+### Development Environment
+
+The [environment](#environment) when you're developing your code. It's accessed through the [CLI](#cli) using `gatsby develop`, and provides extra error reporting and things to help you debug before building for [production](#production-environment).
+
+## E
+
+### ECMAScript
+
+ECMAScript (often referred to as ES) is a specification for scripting languages. [JavaScript](#javascript) is an implementation of ECMAScript. Often developers will use [Babel](#babel) to [compile](#compiler) the latest ECMAScript code into more widely supported JavaScript.
+
+### Environment
+
+The environment that Gatsby runs in. For example, when you are writing your code you probably want as much debugging as possible, but that's undesirable on the live website or app. As such, Gatsby can change its behavior depending on the environment it's in.
+
+Gatsby supports two environments by default, the [development environment](#development-environment) and the [production environment](#production-environment).
+
+### Environment Variables
+
+[Environment Variables](/docs/environment-variables/) allow you to customise the behavior of your app depending on its [environment](#environment). For instance, you may wish to get content from a staging CMS during development and connect to your production CMS when you [build](#build) your site. With environment variables you can set a different URL for each environment.
+
+## F
+
+### Filesystem
+
+The way files are organized. With Gatsby, it means having files in the same place as your website's or app's code instead of pulling data from an external [source](#source)
```

Comment:
```suggestion
The way files are organized. With Gatsby, it means having files in the same place as your website's or app's code instead of pulling data from an external [source](#source). Common filesystem usage in Gatsby includes Markdown content, images, data files, and other assets.
```